pkg/rtc/transport.go

```diff
@@ -84,31 +140,38 @@ type PCTransport struct {
 
 	onFullyEstablished func()
 
-	localDescriptionSent       bool
-	cachedLocalCandidates      []*webrtc.ICECandidate
-	pendingRemoteCandidates    []webrtc.ICECandidateInit
-	debouncedNegotiate         func(func())
-	negotiationPending         map[livekit.ParticipantID]bool
-	onICECandidate             func(c *webrtc.ICECandidate)
-	onOffer                    func(offer webrtc.SessionDescription)
-	onAnswer                   func(offer webrtc.SessionDescription)
-	onRemoteDescriptionSettled func() error
-	onInitialConnected         func()
-	onFailed                   func(isShortLived bool)
-	restartAfterGathering      bool
-	restartAtNextOffer         bool
-	negotiationState           int
-	negotiateCounter           atomic.Int32
-	signalStateCheckTimer      *time.Timer
-	onNegotiationFailed        func()
+	debouncedNegotiate func(func())
+	debouncePending    bool
+
+	onICECandidate      func(c *webrtc.ICECandidate) error
+	onOffer             func(offer webrtc.SessionDescription) error
+	onAnswer            func(offer webrtc.SessionDescription) error
+	onInitialConnected  func()
+	onFailed            func(isShortLived bool)
+	onGetDTX            func() bool
+	onNegotiationFailed func()
 
 	// stream allocator for subscriber PC
 	streamAllocator *sfu.StreamAllocator
 
 	previousAnswer *webrtc.SessionDescription
 
-	preferTCP bool
-
+	preferTCP atomic.Bool
+	isClosed  atomic.Bool
+
+	eventChMu sync.RWMutex
+	eventCh   chan event
+
+	// the following should be accessed only in event processing go routine
+	cacheLocalCandidates      bool
+	cachedLocalCandidates     []*webrtc.ICECandidate
```

Comment:
These should not be required anymore as everything goes through the channel and gets processed in a go routine. But, leaving it in here just for safety.